nextcord/state.py

```diff
@@ -1810,7 +1793,9 @@ def parse_guild_create(self, data) -> None:
 
         # check if it requires chunking
         if self._guild_needs_chunking(guild):
-            asyncio.create_task(self._chunk_and_dispatch(guild, unavailable))
+            task = asyncio.create_task(self._chunk_and_dispatch(guild, unavailable))
+            task.add_done_callback(lambda _t: self._chunk_tasks.pop(guild.id, None))
+            self._chunk_tasks[guild.id] = task
```

Comment:
This has the potential to be a race condition, however extremely unlikely.
```suggestion
            self._chunk_tasks[guild.id] = task
            task.add_done_callback(lambda _t: self._chunk_tasks.pop(guild.id, None))
```

Reply:
I don't think the event loop would context switch in the middle of a synchronous function, but that makes sense to be changed anyway.